### PR TITLE
[Error Handling] - Base PR

### PR DESCRIPTION
**Pull request by W3stside**
Combination of #746 #747 #749 

Not in this PR: all points in #751 (PR comments) - will address these later as they're addons not bug fixes

### CODE REVIEW

File: src/custom/state/price/reducer.ts [hunk 1]
```diff
@@ -23,9 +24,11 @@ export interface PriceInformation {
 }
 
 export interface QuoteInformationObject extends FeeQuoteParams {
-  fee: FeeInformation
-  price: PriceInformation
+  fee?: FeeInformation
+  price?: PriceInformation
+  error?: ApiErrorCodes
   lastCheck: number
+  // TODO: remove
```
Comment (anxolin) [inline src/custom/state/price/reducer.ts:31]: Is this comming in a new PR where we kill the `feeExceedsPrice`?

File: src/custom/state/price/reducer.ts [hunk 2]
```diff
@@ -23,9 +24,11 @@ export interface PriceInformation {
 }
 
 export interface QuoteInformationObject extends FeeQuoteParams {
-  fee: FeeInformation
-  price: PriceInformation
+  fee?: FeeInformation
+  price?: PriceInformation
+  error?: ApiErrorCodes
```
Comment (anxolin) [inline src/custom/state/price/reducer.ts:29]: Are you sure we want to create 1<>1 relation with this error to `ApiErrorCodes` backend codes?

I would think that we just want a new enum with the errors we handle? Maybe I should wait for your next PR and see  how you use this?

**Review by alfetopito — APPROVED** [review]

File: src/custom/state/price/reducer.ts (hunk 1 above)
Comment (W3stside) [reply src/custom/state/price/reducer.ts:31]: yes

File: src/custom/state/price/reducer.ts (hunk 2 above)
Comment (W3stside) [reply src/custom/state/price/reducer.ts:29]: thought about it but don't see any immediate danger doing it this way

Comment (W3stside) [reply src/custom/state/price/reducer.ts:29]: noted in #751

**Comment by anxolin** [comment]
I can review later today again, but is this supposed to be fixed? 

image

**Comment by W3stside** [comment]
> I can review later today again, but is this supposed to be fixed?
> 
> image

no it'll be implemented finally when the 2 prs (this and #735 ) are merged into develop (i have the fix locally)

**Comment by anxolin** [comment]
okok, then yes lets merge

**Pull request merged**